cmd/cdi-cloner/cloner_startup.sh

```diff
@@ -33,15 +40,24 @@ if [ "$obj" == "source" ]; then
     echo "cloner: Starting clone source"
     echo "cloner: creating fifo pipe"
     mkfifo $pipe_dir
-    echo "cloner: creating tarball of the image and redirecting it to $pipe_dir"
-    pushd $image_dir
-	#figure out the size of content in the directory
-    size=$(du -sb . | cut -f1)
-    echo $size
-    #Write the size to the pipe so the other end can read it.
-    echo "$size" >$pipe_dir
-    tar cv --sparse . >$pipe_dir
-    popd
+    if [ "$volumeMode" != "block" ]; then
```

Comment:
this should be:
```bash
    if [ "$volumeMode" == "fs" ]; then
        pushd $image_dir
        #figure out the size of content in the directory
        size=$(du -sb . | cut -f1)
        echo $size
        #Write the size to the pipe so the other end can read it.
        echo "$size" >$pipe_dir
        tar cv --sparse . >$pipe_dir
        popd
    elif [ "$volumeMode" == "block" ]; then
        #figure out the size of the block device
        size=$(lsblk -n -b -o SIZE $image_dir) 
        echo $size
        #Write the size to the pipe so the other end can read it.
        echo "$size" >$pipe_dir
        echo "cloner: writing the image to $pipe_dir"
        dd if=$image_dir bs=4M of=$pipe_dir 
    fi
```
This way you are explicitly checking for the right mode, and for BOTH modes you are passing in the size in first, and then you are writing the actual data. Also the block size of 4M seems awfully large. Now we can have a huge long debate over the best size of a block with DD, but going a quick google search it appears the optimal size to be somewhere in the 32-128k range. So I guess 64k seems like a decent middle ground.

Reply:
Won't 'dd if=$image_dir bs=4M of=$pipe_dir' overwrite the existing $pipe_dir that already contains the size? will it append?

Reply:
Its a pipe, it acts like a stream, anything that is written to it, will get appended.